File: S2P2_herencia/P2A4Empleado/P2A4C++/Empleado.cpp
#include "Empleado.h"
#include <sstream>

Empleado::Empleado(const std::string& nombre, const std::string& apellido, const std::string& nss)
    : primerNombre(nombre), apellidoPaterno(apellido), numeroSeguroSocial(nss) {}

void Empleado::establecerPrimerNombre(const std::string& nombre) {
    primerNombre = nombre;
}

std::string Empleado::obtenerPrimerNombre() const {
    return primerNombre;
}

void Empleado::establecerApellidoPaterno(const std::string& apellido) {
    apellidoPaterno = apellido;
}

std::string Empleado::obtenerApellidoPaterno() const {
    return apellidoPaterno;
}

void Empleado::establecerNumeroSeguroSocial(const std::string& nss) {
    numeroSeguroSocial = nss;
}

std::string Empleado::obtenerNumeroSeguroSocial() const {
    return numeroSeguroSocial;
}

std::string Empleado::toString() const {
    std::ostringstream output;
    output << obtenerPrimerNombre() << " " << obtenerApellidoPaterno()
           << "\nnumero de seguro social: " << obtenerNumeroSeguroSocial();
    return output.str();
}

File: 1POO/Semestre 1 - 2/S2P2_herencia/P2A4Empleado/P2A4Python/Empleado.py
from abc import ABC, abstractmethod

class Empleado(ABC):
    def __init__(self, nombre, apellido, nss):
        self._primerNombre = nombre
        self._apellidoPaterno = apellido
        self._numeroSeguroSocial = nss

    def establecerPrimerNombre(self, nombre):
        self._primerNombre = nombre

    def obtenerPrimerNombre(self):
        return self._primerNombre

    def establecerApellidoPaterno(self, apellido):
        self._apellidoPaterno = apellido

    def obtenerApellidoPaterno(self):
        return self._apellidoPaterno

    def establecerNumeroSeguroSocial(self, nss):
        self._numeroSeguroSocial = nss

    def obtenerNumeroSeguroSocial(self):
        return self._numeroSeguroSocial

    def __str__(self):
        return f"{self.obtenerPrimerNombre()} {self.obtenerApellidoPaterno()}\nnumero de seguro social: {self.obtenerNumeroSeguroSocial()}"

    @abstractmethod
    def Ingresos(self):
        pass
